Website/Providers/DataProviders/SqlDataProvider/09.02.01.SqlDataProvider

```diff
@@ -102,6 +102,115 @@ ELSE
 		AND    ModuleId = @ModuleId
 GO
 
+
+/*** [DNN-18327]:[ISSUE-2096] Replace underscores with spaces. ***/
+IF  EXISTS (SELECT * FROM sys.objects WHERE object_id = OBJECT_ID(N'{databaseOwner}[{objectQualifier}AddPropertyDefinition]') AND type in (N'P', N'PC'))
+	DROP PROCEDURE {databaseOwner}[{objectQualifier}AddPropertyDefinition]
+GO
+
+CREATE PROCEDURE {databaseOwner}[{objectQualifier}AddPropertyDefinition]
+	@PortalId int,
+	@ModuleDefId int,
+	@DataType int,
+	@DefaultValue nvarchar(max),
+	@PropertyCategory nvarchar(50),
+	@PropertyName nvarchar(50),
+	@ReadOnly bit,
+	@Required bit,
+	@ValidationExpression nvarchar(2000),
+	@ViewOrder int,
+	@Visible bit,
+    @Length int,
+    @DefaultVisibility int,
+	@CreatedByUserID int
+
+AS
+	DECLARE @PropertyDefinitionId int
+
+	SET @PropertyName = Replace(LTrim(RTrim(Replace(@PropertyName,'_',' '))), '_', ' ')
```

Comment:
Is it necessary to to `REPLACE` twice?

Reply:
Unfortunately yes. Because of the test cases To work with both cases: "_Hire Date_" and "  Hire_Date  ".